pkg/testworkflows/testworkflowexecutor/scheduler.go

```diff
@@ -260,24 +272,107 @@ func (s *scheduler) Schedule(ctx context.Context, sensitiveDataHandler Sensitive
 			return ch, err
 		}
 		for _, w := range list {
-			targets := execution.Targets
-
-			if isEmptyTargets(targets) && w.Spec.Execution != nil && w.Spec.Execution.Target != nil {
-				target := commonmapper.MapTargetApiToGrpc(w.Spec.Execution.Target)
-				targets = []*cloud.ExecutionTarget{target}
-			}
-
 			intermediateSelectors = append(intermediateSelectors, &cloud.ScheduleExecution{
 				Selector:      &cloud.ScheduleResourceSelector{Name: w.Name},
-				Targets:       targets,
+				Targets:       execution.Targets,
 				Config:        execution.Config,
 				ExecutionName: execution.ExecutionName, // TODO: what to do when execution name is configured, but multiple requested?
 				Tags:          execution.Tags,
 			})
 		}
 	}
 
+	// Determine and normalise the target for the scheduled executions
+	//
+	// Priority from high to low:
+	// - The target found directly on the scheduled request.
+	// - The target found on the Workflow Custom Resource.
+	// - The target found on any of the workflow's used templates. Throws an error if multiple define targets.
+	for i, _ := range intermediateSelectors {
+		exec := intermediateSelectors[i]
+		targets := exec.Targets
+
+		// Check the request for targets
+		if !isEmptyTargets(targets) {
+			continue
+		}
+
+		// Check the workflow for targets
+		workflows, err := testWorkflows.Get(exec.Selector)
+		if err != nil {
+			close(ch)
+			return ch, err
+		}
+
+		if len(workflows) == 0 {
+			close(ch)
+			return ch, errors.New("workflow not found")
+		}
+		w := workflows[0]
+		if w.Spec != nil && w.Spec.Execution != nil && w.Spec.Execution.Target != nil {
+			target := commonmapper.MapTargetApiToGrpc(w.Spec.Execution.Target)
+			exec.Targets = []*cloud.ExecutionTarget{target}
+			continue
+		}
+
+		// Check the templates for targets
+		var target *testkube.ExecutionTarget
+		var targetTemplateName string
+		templateNames := getTemplateNames(w)
+		for _, templateName := range templateNames {
```

Comment:
Looking at your actual usage here I don't even think there is much value in the separate function, especially as it is valid to `range` over a `nil` slice (produces no iterations). Instead you can just simply:
```suggestion
		if w.Spec == nil {
			continue
		}
		for _, template := range w.Spec.Use {
			templateName := template.Name
```